packages/rum/src/index.js

```diff
@@ -26,20 +26,27 @@
 import {
   createServiceFactory,
   bootstrap,
-  isBrowser
+  isBrowser,
+  CONFIG_SERVICE,
+  TRANSACTION_SERVICE
 } from '@elastic/apm-rum-core'
 import ApmBase from './apm-base'
 
 /**
  * Use a single instance of ApmBase across all instance of the agent
- * including the instanes used in framework specific integrations
+ * including the instances used in framework specific integrations
  */
 function getApmBase() {
   if (isBrowser && window.elasticApm) {
     return window.elasticApm
   }
-  const enabled = bootstrap()
   const serviceFactory = createServiceFactory()
+  const [configService, transactionService] = serviceFactory.getService([
+    CONFIG_SERVICE,
+    TRANSACTION_SERVICE
+  ])
+
+  const enabled = bootstrap(configService, transactionService)
```

Comment:
the reason why bootstrap was called first before creating service factory is that we need to account for when the time the agent got initialized, we might need to split this with installing page visibility handlers. if we need service factories to be created.

Reply:
Your comment just make me realise that a better place to add the page visibility handlers could be inside ApmBase.init otherwise we would be adding listeners even with the agent not initialised